### DISCUSSION TITLE
Modify WithPagination.php

### DISCUSSION THREAD

**Discussion by HDVinnie** [livewire/livewire#3373] [3. Feature ideas]
Wondering if it is warranted to create a PR modifying WithPagination.php by adding an `$this->emit('gotoTop');` to the `previousPage`, `nextPage` and `gotoPage` functions so end users like myself can easily hook into it for pagination scrolling.

```
        <script nonce="{{ Bepsvpt\SecureHeaders\SecureHeaders::nonce('script') }}">
            Livewire.on('gotoTop', () => {
                window.scrollTo({
                    top: 15,
                    left: 15,
                    behaviour: 'smooth'
                })
            })
        </script>
```

This way when a user goes to the next page it will scroll back to top of page which is expected behavior.

**Comment by HDVinnie** [comment]
Any news?

**Comment by joshhanley** [comment]
@HDVinnie you could actually just modify `setPage` method to emit an event whenever a page changes (rather than all 3 methods).

I probably wouldn't call it `'gotToTop'`, I'd make it more related to the event that happened, like `'paginationChanged'` then anyone can listen for that and do what they desire, like scroll to the top in your case.

I don't see any issue with that as a PR.

But if you wanted to use that now, you can override the set page method in your component to trigger that

```php
public function setPage($page)
{
    $this->page = $page;
    
    $this->emit('paginationChanged');
}
```

Hope this helps!
